src/main/java/io/vertx/core/net/KeyStoreOptionsBase.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2011-2019 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.net;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.impl.VertxInternal;
+import io.vertx.core.net.impl.KeyStoreHelper;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509KeyManager;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Internal base class of {@code KeyStore} based options.
+ *
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ * @author <a href="http://tfox.org">Tim Fox</a>
+ */
+class KeyStoreOptionsBase implements KeyCertOptions, TrustOptions {
+
+  private KeyStoreHelper helper;
+  private String type;
+  private String password;
+  private String path;
```

Comment:
@vietj Hi Julien - it looks good, can this base class include `provider` now ?

Reply:
yes, but how can we test it ?

Reply:
I think you can use "jce" as provider to load a jceks file. Although this provider is always present if I'm not mistaken.

Reply:
ok, I'll have a look, perhaps we can write an integration test with a mock provider that delegates to JCE ?

Reply:
it is actually used `KeyStore.getInstance("JKS").getProvider().getName()` that returns the _SUN_ value